build-aux/Jenkinsfile.full

```diff
@@ -486,21 +457,92 @@ pipeline {
           } // post
         } // stage
 
+	/*
+	 *
+	 * This is just taking too long to run. Right now, the khash tests are timing out
+	 * on the IBM servers:
+	 *
+         * [2019-12-31T20:58:48.704Z]   khash randomized test
+         * [2019-12-31T20:59:04.869Z]     khash_test:103: randomized_test_ (State matches dict implementation)...*timed out*
```

Comment:
@davisp FYI. We don't need emulated test runs to pass, but this usually passes for me locally before timeout. Is this a test we can improve at all?

Reply:
Its a long test by design really. The emulation is probably just slowing things down far enough to be an issue. I have a somewhat related branch [1] where I was poking at shortening some of these property based tests when we're running a PR. I was waffling between "run shorter on PR" and "only run long ones on nightly builds" which are slightly different. The PR approach I landed on wouldn't help here. But I was also meaning to ask if we might want to add a nightly "Long test" build that bumps a lot of the test sizes that we could run nightly or some such;

[1] https://github.com/apache/couchdb/commit/f290788be4beee412438580e17bf9016851c1f6f

Reply:
Hey thanks Paul.

The current approach in this PR is to run the full test+packages build on all the platforms we can support. That could optionally run a longer set of tests if you like.

Right now in this PR that will run on `master` and any release branch after any change to that branch. It's not nightly, but it is also not a PR-blocking run. I think that meets our needs?

Reply:
Sorry, I don't think I was clear enough earlier. What I was asking/proposing was that we create a new type of nightly build that is something like "Nightly Soak Test" where we jack up all of the values for our property/long unit tests that runs once a day on only a single platform.

My current approach is the opposite in that we tune down some parameters on PRs only as opposed to everything except the soak test build.

Also to be clear, these long running tests are actually not entirely supposed to be "unit tests" per se. The property based testing is (in my experience) supposed to *generate* unit tests. The workflow would basically be something like, design your property, run it for many hours, any failures are added as regression tests. And the "run for hours" part is executed before major events like large PRs and releases and so on.

And alternatively, something else we could do is pull all of those tests out into a separate manual build that *only* runs the property tests and actually does run them for hours on a manual button push.